pkg/interceptors/interceptors.go

```diff
@@ -17,27 +17,83 @@ limitations under the License.
 package interceptors
 
 import (
+	"errors"
+	"fmt"
 	"net/http"
+	"time"
 
 	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 )
 
-// Interceptor is the interface that all interceptors implement.
-type Interceptor interface {
-	ExecuteTrigger(req *http.Request) (*http.Response, error)
+// WebhookSecretStore provides cached lookups of k8s secrets, backed by a Reflector.
+type WebhookSecretStore interface {
+	Get(sr triggersv1.SecretRef) ([]byte, error)
+}
+
+type webhookSecretStore struct {
+	store                  cache.Store
+	eventListenerNamespace string
+}
+
+// NewWebhookSecretStore provides cached lookups of k8s secrets, backed by a Reflector.
+func NewWebhookSecretStore(cs kubernetes.Interface, ns string, resyncInterval time.Duration, stopCh <-chan struct{}) WebhookSecretStore {
+	secretsClient := cs.CoreV1().Secrets(metav1.NamespaceAll)
+	store := cache.NewStore(func(obj interface{}) (string, error) {
+		secret, ok := obj.(corev1.Secret)
+		if !ok {
+			return "", errors.New("Object is not a secret")
```

Comment:
```suggestion
			return "", errors.New("object is not a secret")
```
I think the convention is to use lower-cased errors so you can wrap errors without uneven capitalisation.